src/libsyntax/parse/parser.rs

```diff
@@ -1523,8 +1523,17 @@ impl<'a> Parser<'a> {
             let pat = self.parse_pat(Some("argument name"))?;
 
             if let Err(mut err) = self.expect(&token::Colon) {
-                self.argument_without_type(&mut err, pat, require_name, is_trait_item);
-                return Err(err);
+                if let Some(ident) = self.argument_without_type(
```

Comment:
--> diagnostics.rs ?

Reply:
This is already a call to a method in `diagnostics.rs`